File: app/src/main/java/com/example/vladimir/easyenglishlearn/ExerciseActivity.java
package com.example.vladimir.easyenglishlearn;

import android.content.Intent;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.vladimir.easyenglishlearn.exercises.WordConstructorFragment;
import com.example.vladimir.easyenglishlearn.exercises.WordQuizFragment;
import com.example.vladimir.easyenglishlearn.model.Word;

import java.util.ArrayList;

import static com.example.vladimir.easyenglishlearn.Constants.EXERCISE_TYPE;
import static com.example.vladimir.easyenglishlearn.Constants.SELECTED_WORDS;
import static com.example.vladimir.easyenglishlearn.Constants.TRANSLATION_DIRECTION;
import static com.example.vladimir.easyenglishlearn.Constants.WORD_CONSTRUCTOR;

public class ExerciseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_exercise);

        Intent intent = getIntent();
        String exerciseType = intent.getStringExtra(EXERCISE_TYPE);
        ArrayList<Word> selectedWordList = intent.getParcelableArrayListExtra(SELECTED_WORDS);
        boolean translationDirection = intent.getBooleanExtra(TRANSLATION_DIRECTION, true);

        FragmentManager fm = getSupportFragmentManager();
        Fragment fragment = fm.findFragmentById(R.id.exercise_fragment_container);
        if (fragment == null) {
            switch (exerciseType) {
                case WORD_CONSTRUCTOR:
                    fragment = WordConstructorFragment
                            .newInstance(selectedWordList, translationDirection);
                    break;
                default:
                    fragment = WordQuizFragment
                            .newInstance(selectedWordList, translationDirection);
            }

            fm.beginTransaction()
                    .add(R.id.exercise_fragment_container, fragment)
                    .commit();
        }
    }
}

File: app/src/main/java/com/vladimir_x/easyenglishlearn/ui/ExerciseActivity.kt
package com.vladimir_x.easyenglishlearn.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.vladimir_x.easyenglishlearn.Constants
import com.vladimir_x.easyenglishlearn.R
import com.vladimir_x.easyenglishlearn.databinding.ActivityExerciseBinding
import com.vladimir_x.easyenglishlearn.ui.exercises.ConstructorFragment
import com.vladimir_x.easyenglishlearn.ui.exercises.QuizFragment
import com.vladimir_x.easyenglishlearn.ui.model.WordUI
import com.vladimir_x.easyenglishlearn.ui.extension.getParcelableArrayList
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class ExerciseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityExerciseBinding.inflate(layoutInflater)
        setContentView(binding.root)
        val intent = intent
        val exerciseType = intent.getStringExtra(Constants.EXERCISE_TYPE)
        val selectedWordList =
            intent.getParcelableArrayList<WordUI>(Constants.SELECTED_WORDS) as ArrayList<WordUI>
        val translationDirection = intent.getBooleanExtra(Constants.TRANSLATION_DIRECTION, true)
        val fm = supportFragmentManager
        var fragment = fm.findFragmentById(binding.exerciseFragmentContainer.id)
        if (fragment == null) {
            fragment = when (exerciseType) {
                Constants.WORD_CONSTRUCTOR -> ConstructorFragment
                    .newInstance(selectedWordList, translationDirection)
                else -> QuizFragment
                    .newInstance(selectedWordList, translationDirection)
            }
            fm.beginTransaction()
                .add(R.id.exercise_fragment_container, fragment)
                .commit()
        }
    }
}
